File: Release/WindowsPhone7/PhoneToolkitSample/Data/PeopleByFirstName.cs
// (c) Copyright Microsoft Corporation.
// This source is subject to the Microsoft Public License (Ms-PL).
// Please see http://go.microsoft.com/fwlink/?LinkID=131993 for details.
// All other rights reserved.

using System.Collections.Generic;

namespace PhoneToolkitSample.Data
{
    public class PeopleByFirstName : List<PeopleInGroup>
    {
        private static readonly string Groups = "#abcdefghijklmnopqrstuvwxyz";

        private Dictionary<int, Person> _personLookup = new Dictionary<int, Person>();

        public PeopleByFirstName()
        {
            List<Person> people = new List<Person>(AllPeople.Current);
            people.Sort(Person.CompareByFirstName);

            Dictionary<string, PeopleInGroup> groups = new Dictionary<string, PeopleInGroup>();

            foreach (char c in Groups)
            {
                PeopleInGroup group = new PeopleInGroup(c.ToString());
                this.Add(group);
                groups[c.ToString()] = group;
            }

            foreach (Person person in people)
            {
                groups[Person.GetFirstNameKey(person)].Add(person);
            }
        }
    }
}

File: Release/WindowsPhone7/PhoneToolkitSample/Data/PeopleByFirstName.vb
' (c) Copyright Microsoft Corporation.
' This source is subject to the Microsoft Public License (Ms-PL).
' Please see http://go.microsoft.com/fwlink/?LinkID=131993 for details.
' All other rights reserved.

Namespace Data

    Public Class PeopleByFirstName
        Inherits List(Of PeopleInGroup)
        Private Shared ReadOnly Groups As String = "#abcdefghijklmnopqrstuvwxyz"

        Private _personLookup As New Dictionary(Of Integer, Person)

        Public Sub New()
            Dim people As New List(Of Person)(AllPeople.Current)
            people.Sort(AddressOf Person.CompareByFirstName)

            Dim groups As New Dictionary(Of String, PeopleInGroup)

            For Each c In PeopleByFirstName.Groups
                Dim group As New PeopleInGroup(c.ToString())
                Me.Add(group)
                groups(c.ToString()) = group
            Next c

            For Each _person In people
                groups(Person.GetFirstNameKey(_person)).Add(_person)
            Next _person
        End Sub
    End Class

End Namespace
